packages/itwin/grouping-mapping-widget/src/widget/components/Grouping.tsx

```diff
@@ -383,35 +412,141 @@ export const Groupings = ({ mapping, goBack }: GroupsTreeProps) => {
             title={mapping.mappingName ?? ""}
             disabled={isLoading || isLoadingQuery}
             returnFn={async () => {
-              clearEmphasizedElements();
+              clearEmphasizedOverriddenElements();
               await goBack();
             }}
           />
-          <div className='groups-container'>
-            <Button
-              startIcon={
-                isLoadingQuery ? <ProgressRadial size="small" indeterminate /> : <SvgAdd />
-              }
-              styleType='high-visibility'
-              disabled={isLoadingQuery}
-              onClick={() => addGroup()}
-            >
-              {isLoadingQuery ? "Loading Group(s)" : "Add Group"}
-            </Button>
-            <Table<GroupType>
-              data={groups}
-              density='extra-condensed'
-              columns={groupsColumns}
-              emptyTableContent='No Groups available.'
-              isSortable
-              isSelectable
-              onSelect={onSelect}
-              isLoading={isLoading}
-              isRowDisabled={() => isLoadingQuery}
-              stateReducer={tableStateReducer}
-              useControlledState={controlledState}
-            />
-          </div>
+          <Surface className="groups-container">
+            <div className="toolbar">
+              <Button
+                startIcon={
+                  isLoadingQuery ? (
+                    <ProgressRadial size="small" indeterminate />
+                  ) : (
+                    <SvgAdd />
+                  )
+                }
+                styleType="high-visibility"
+                disabled={isLoadingQuery}
+                onClick={addGroup}
+              >
+                {isLoadingQuery ? "Loading Group(s)" : "Add Group"}
+              </Button>
+              <ButtonGroup className="toolbar-buttons">
+                <ToggleSwitch
+                  label="Color by Group"
+                  labelPosition="left"
+                  className="group-view-icon toggle"
+                  disabled={isLoadingQuery}
+                  checked={showGroupColor}
+                  onChange={toggleGroupColor}
+                ></ToggleSwitch>
```

Comment:
I think the buttons should be disabled when not showing group colors. `{isLoadingQuery || !showGroupColor}`